src/main/java/org/opentripplanner/middleware/controllers/api/ApiController.java

```diff
@@ -140,18 +145,35 @@ public void bind(final SparkSwagger restApi) {
     }
 
     /**
-     * HTTP endpoint to get multiple entities.
+     * HTTP endpoint to get multiple entities based on the user permissions
      */
+    // FIXME Maybe better if the user check (and filtering) was done in a pre hook?
+    // FIXME Will require further granularity for admin
     private List<T> getMany(Request req, Response res) {
-        return persistence.getAll();
+
+        Auth0UserProfile requestingUser = getUserFromRequest(req);
+        if (isUserAdmin(requestingUser)) {
+            return persistence.getAll();
+        } else if (persistence.clazz == OtpUser.class) { 
+            Bson filter = Filters.eq("_id", requestingUser.otpUser.id);
+            return persistence.getFiltered(filter);
+        } else {
+            Bson filter = Filters.eq("userId", requestingUser.otpUser.id);
+            return persistence.getFiltered(filter);
+        }
     }
 
     /**
      * HTTP endpoint to get one entity specified by ID.
      */
     private T getOne(Request req, Response res) {
+        Auth0UserProfile requestingUser = Auth0Connection.getUserFromRequest(req);
         String id = getIdFromRequest(req);
-        return getObjectForId(req, id);
+        T object = getObjectForId(req, id);
+        if (!object.canBeManagedBy(requestingUser)) {
```

Comment:
I think the name `canBeManagedBy` implies that the requesting user can update or delete the entity. I'm not sure if it applies to viewing the entity. Do we need a separate method for `getOne`?

Reply:
@landonreed I think you are right. I'm not sure how a separate method that would tie-in with the controller 'get one entity' endpoint. Perhaps the way forward is along the same lines as getMany?

Reply:
Yea, I think you're right. If the object can be managed by the user, the user is obviously authorized. Else, the requesting userId must match the userId attached to the requested object.

Reply:
@landonreed - Thanks. Will update...

Reply:
@landonreed - After testing, it is the responsibility of the overriding class to define access to it's object. I have updated the method comment to clarify this. Therefore, I don't think the method should be changed.